File: articles/causal-algorithms-and-intervention-modeling/typescript/intervention_policy.ts
function decision(score: number, threshold: number): string {
  return score >= threshold ? "act" : "monitor";
}

const baseline = 0.42;
const intervention = 0.57;
console.log(`estimated_effect=${(intervention - baseline).toFixed(6)}`);
console.log(`baseline_decision=${decision(0.53, 0.55)}`);
console.log(`new_threshold_decision=${decision(0.53, 0.50)}`);

File: articles/causal-algorithms-and-intervention-modeling/go/intervention_policy.go
package main

import "fmt"

func decision(score float64, threshold float64) string {
	if score >= threshold { return "act" }
	return "monitor"
}

func main() {
	baseline := 0.42
	intervention := 0.57
	fmt.Printf("estimated_effect=%.6f
", intervention-baseline)
	fmt.Println("baseline_decision=", decision(0.53, 0.55))
	fmt.Println("new_threshold_decision=", decision(0.53, 0.50))
}
